KITE-Extras/src/main/java/io/cosmosoftware/kite/instrumentation/Instance.java

```diff
@@ -20,37 +8,59 @@
 
 /**
  * The type Instance: represents a selenium node or a gateway for network instrumentation.
+ *     {
+ *       "_id": "gw1",
+ * 	  "type": "gateway",
+ *       "name": "Alice Gateway",
+ *       "ipAddress": "11.222.33.238",
+ *       "username": "gw-user",
+ *       "password": "optional",
+ *       "keyFilePath": "file://path/to/privatekey.pem",
+ * 	  "nit0": "eth0",
+ * 	  "nit1": "eth1",
+ * 	  "nit2": "eth2",
+ *     },
  */
 public class Instance {
 
-  private String id;
+  private final String id;
 
-  private String username;
+  private final String username;
 
-  private String name;
+  private final String name;
 
-  private String ipAddress;
+  private final String ipAddress;
 
-  private String keyFilePath;
+  private final String keyFilePath;
+
+  private final String type;
+
+  private final String password;
+
+  private final String nit0;
+
+  private final String nit1;
+
+  private final String nit2;
 
-  /**
-   * Instantiates a new Instance.
-   */
-  public Instance() {
-    super();
-  }
 
   /**
    * Instantiates a new Instance.
    *
    * @param jsonObject the json object
    */
   public Instance(JsonObject jsonObject) {
-    this.id = jsonObject.getString("_id");
-    this.ipAddress = jsonObject.getString("ipAddress");
-    this.username = jsonObject.getString("username");
-    this.name = jsonObject.getString("name");
-    this.keyFilePath = jsonObject.getString("keyFilePath");
+
+      this.id = jsonObject.getString("_id");
```

Comment:
if the key is mandatory, then try {
missingKey = "_id";
this.id = jsonObject.getString("_id");
missingKey = "ipAddress";
this.id = jsonObject.getString("ipAddress");
[...]

} catch(NullPointerException e) {
  throw
}

if the key is not mandatory
this.password = jsonObject.getString("password", null);